code/modules/mapping/ruins.dm

```diff
@@ -8,37 +8,41 @@
 		var/height_border = TRANSITIONEDGE + SPACERUIN_MAP_EDGE_PAD + round(height / 2)
 		var/turf/central_turf = forced_turf ? forced_turf : locate(rand(width_border, world.maxx - width_border), rand(height_border, world.maxy - height_border), z)
 		var/valid = TRUE
+		var/list/affected_turfs = get_affected_turfs(central_turf,1)
+		var/list/affected_areas = list()
 
-		for(var/turf/check in get_affected_turfs(central_turf,1))
-			var/area/new_area = get_area(check)
-			valid = FALSE // set to false before we check
+		for(var/turf/check in affected_turfs)
+			// Use assoc lists to move this out, it's easier that way
 			if(check.turf_flags & NO_RUINS)
+				valid = FALSE // set to false before we check
 				break
-			for(var/type in allowed_areas)
-				if(istype(new_area, type)) // it's at least one of our types so it's whitelisted
-					valid = TRUE
-					break
-			if(!valid)
+			var/area/new_area = get_area(check)
+			affected_areas[new_area] = TRUE
+
+		// This is faster yes. Only BARELY but it is faster
+		for(var/area/affct_area as anything in affected_areas)
+			if(!allowed_areas_typecache[affct_area.type])
+				valid = FALSE
 				break
 
 		if(!valid)
 			continue
 
 		testing("Ruin \"[name]\" placed at ([central_turf.x], [central_turf.y], [central_turf.z])")
 
-		for(var/i in get_affected_turfs(central_turf, 1))
-			var/turf/T = i
-			for(var/obj/structure/spawner/nest in T)
-				qdel(nest)
-			for(var/mob/living/simple_animal/monster in T)
-				qdel(monster)
-			for(var/obj/structure/flora/plant in T)
-				qdel(plant)
+		if(clear_below)
+			for(var/turf/T as anything in affected_turfs)
+				for(var/obj/structure/spawner/nest in T)
+					qdel(nest)
+				for(var/mob/living/simple_animal/monster in T)
+					qdel(monster)
+				for(var/obj/structure/flora/plant in T)
+					qdel(plant)
```

Comment:
wat

i figure this isnt hot but wat

Reply:
oh yeah uh
I mean it's kinda hard to say. I'd have to profile for which is actually faster, I'm not sure

Reply:
Will profile it

Reply:
![image](https://user-images.githubusercontent.com/58055496/189521078-de01db99-b912-4fdf-ac3a-0366411e0ef8.png)
technically faster yes

Reply:
what.

Reply:
i bet just using istype rather than jumping through hoops is faster but whatever this is cold anyway